lib/crowbar/client/request/upgrade.rb

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2016, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
```

Comment:
The same as mentioned for backup.rb. Now I see the modules being re-repeated again. The data which is different could be fetched from a configuration file or encoded into a simple DS. Also the other comments apply to this file and of course the following files that I can see coming already. It may not be extremely crucial to do this but should be considered.